unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,217 @@
+                   redis_client_auto_register._on_unregister,
+               f"{AREA_ID}/*": redis_client_auto_register._on_event_or_response,
+               "aggregator_response": redis_client_auto_register._aggregator_response_callback})
+
+    def test_aggregator_response_callback(self, redis_client_auto_register):
+        """checks whether the transaction id buffer is popping out and returning the empty list"""
+        message = {"data": {"transaction_id": f"{TRANSACTION_ID}"}}
+        message["data"] = json.dumps(message["data"])
+        redis_client_auto_register._transaction_id_buffer.append(TRANSACTION_ID)
+        redis_client_auto_register._aggregator_response_callback(message)
+        assert redis_client_auto_register._transaction_id_buffer == []
+
+    def test_check_buffer_message_matching_command_and_id(self, redis_client_auto_register):
+        """checks the buffer message matching the command and id and returns as None"""
+        message = {"data": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._blocking_command_responses \
+            = {"register": {"transaction_id": f"{TRANSACTION_ID}"}}
+        assert redis_client_auto_register \
+                   ._check_buffer_message_matching_command_and_id(message["data"]) is None
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self, redis_client_auto_register):
+        """checks buffer message matching command and id does
+           contain transaction id throws exception"""
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match="The answer message does not contain a valid "
+                                 "\'transaction_id\' member."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(message)
+
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(self, redis_client_auto_register):
+        """checks buffer message matching command and id throws exception"""
+        message = {"data": {"transaction_id": TRANSACTION_ID}}
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register \
+                ._check_buffer_message_matching_command_and_id(message["data"])
+
+    def test_register(self, redis_client_auto_register):
+        """checks whether the redis.db.publish is called with correct arguments"""
+        data = {"name": AREA_ID, "transaction_id": AREA_ID}
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        redis_client_auto_register.redis_db.publish \
+            .assert_called_once_with(f"{AREA_ID}/register_participant", json.dumps(data))
+
+    def test_register_blocking_true_throws_exception(self, redis_client_auto_register):
```

Comment:
Here the cause it not that `is_blocking=True`, but that there is no d3a running and is connected via redis. Please include this at least in the docstring. Thanks!